src/EnergyPlus/WeatherManager.cc

```diff
@@ -2309,33 +2309,35 @@ namespace WeatherManager {
         if (state.dataEnvrn->EMSBeamSolarRadOverrideOn) state.dataEnvrn->BeamSolarRad = state.dataEnvrn->EMSBeamSolarRadOverrideValue;
         state.dataEnvrn->LiquidPrecipitation =
             state.dataWeatherManager->TodayLiquidPrecip(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay) / 1000.0; // convert from mm to m
-        if ((state.dataEnvrn->RunPeriodEnvironment) && (!state.dataGlobal->WarmupFlag)) {
-            int month = state.dataEnvrn->Month;
-            state.dataWaterData->RainFall.MonthlyTotalPrecInWeather.at(month - 1) += state.dataEnvrn->LiquidPrecipitation * 1000.0;
-            if ((state.dataEnvrn->LiquidPrecipitation > 0) && (state.dataGlobal->TimeStep == 1)) {
-                state.dataWaterData->RainFall.numRainyHoursInWeather.at(month - 1) += 1;
-            }
-        }
 
         // if using epw precipitation, throw warning
```

Comment:
Looking at this now, it would be clearer (and save a couple of if blocks) to move this (lines 2313 thru 2332) into `WaterManager::UpdatePrecipitation. Review that function for unnecessary if blocks as well.

Reply:
I combined this with the if statement in UpdatePrecipitation